pkg/internalapis/featurestates/featurestates.go

```diff
@@ -0,0 +1,399 @@
+			log.Infof("performing retry operations on failed feature states CRs")
+
+			for namespace, featurestatesSpec := range pendingCRUpdatesObj.pendingCRUpdate {
+				if featurestatesSpec != nil {
+					// Check if namespace for the CR is still present on the cluster
+					_, err := k8sClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+					if err != nil {
+						if apierrors.IsNotFound(err) {
+							// if Namespace is not present, mark pending task as done
+							pendingCRUpdatesObj.pendingCRUpdate[namespace] = nil
+							continue
+						} else {
+							log.Errorf("failed to check if namespace: %q is present on the cluster", namespace)
+							continue
+						}
+					}
+					// check if CR is present on the namespace
+					featurestateCR := &featurestatesv1alpha1.CnsCsiSvFeatureStates{}
+					err = controllerRuntimeClient.Get(ctx, client.ObjectKey{Name: SVFeatureStateCRName, Namespace: namespace}, featurestateCR)
+					if err != nil {
+						if !apierrors.IsNotFound(err) {
+							// attempt to Create the CR
+							err = controllerRuntimeClient.Create(ctx, featurestatesSpec)
+							if err != nil {
+								log.Errorf("failed to create cnsCsiSvFeatureStates CR instance in the namespace: %q, Err: %v", featurestatesSpec.Namespace, err)
+							} else {
+								log.Infof("created cnsCsiSvFeatureStates CR instance in the namespace: %q", featurestatesSpec.Namespace)
+							}
+						} else {
+							log.Errorf("failed to get cnsCsiSvFeatureStates CR instance from the namespace: %q, Err: %v", featurestatesSpec.Namespace, err)
+						}
+					} else {
+						// Attempt to Update CR
+						featurestateCR.Spec.FeatureStates = featurestatesSpec.Spec.FeatureStates
+						err = controllerRuntimeClient.Update(ctx, featurestateCR)
+						if err != nil {
+							log.Errorf("failed to update cnsCsiSvFeatureStates CR instance in the namespace: %q, Err: %v", featurestatesSpec.Namespace, err)
+						} else {
+							log.Infof("updated cnsCsiSvFeatureStates CR instance in the namespace: %q", featurestatesSpec.Namespace)
+						}
+					}
+					// if no error observed, mark pending task as done
+					if err == nil {
+						pendingCRUpdatesObj.pendingCRUpdate[namespace] = nil
+					}
+				}
+			}
+		}()
+		time.Sleep(crUpdateRetryInterval)
+	}
+}
+
+// updateFeatureStatesCRInAllWorkLoadNamespaces helps update latest featureStates to all workload namespaces
+func updateFeatureStatesCRInAllWorkLoadNamespaces(ctx context.Context, featurestates []featurestatesv1alpha1.FeatureState) error {
+	pendingCRUpdatesObj.lock.Lock()
+	defer pendingCRUpdatesObj.lock.Unlock()
+
+	log := logger.GetLogger(ctx)
+	log.Info("updating feature states in all workload namespaces")
+
+	// retrieving all workload namespaces
+	namespaces, err := k8sClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
+		LabelSelector: WorkLoadNamespaceLabelKey,
+	})
+	if err != nil {
+		log.Errorf("failed to get workload namespaces. Err: %v", err)
```

Comment:
failed to get workload namespaces ==> `failed to list workload namespaces`

Reply:
done.